packages/hydrogen/src/UIImage.web.tsx

```diff
@@ -0,0 +1,155 @@
+    ColorValue,
+    Image as RNImage,
+    ImageProps,
+    ImageStyle,
+    LayoutChangeEvent,
+    LayoutRectangle,
+    StyleProp,
+    StyleSheet,
+    View,
+} from 'react-native';
+import { ColorVariants, useTheme } from './Colors';
+
+const useImageDimensions = (style: StyleProp<ImageStyle>, source: any) => {
+    return React.useMemo(() => {
+        let width: number | string | undefined = undefined;
+        let height: number | string | undefined = undefined;
+        const flatStyle = StyleSheet.flatten(style);
+        if (flatStyle) {
+            if (flatStyle.width) {
+                width = flatStyle.width;
+            }
+            if (flatStyle.height) {
+                height = flatStyle.height;
+            }
+        }
+        if (!width && source && source.width) {
+            width = source.width;
+        }
+        if (!height && source && source.height) {
+            height = source.height;
+        }
+        return {
+            width,
+            height,
+        };
+    }, [style, source && source.uri]);
+};
+
+export type UIImageProps = ImageProps & {
+    /**
+     * tintColor in some cases don't work on Safary with RNImage.
+     * Here is the issue: https://github.com/necolas/react-native-web/issues/1914
+     * Hence passing this prop we force to use canvas
+     */
+    tintColor?: ColorVariants | null;
+};
+
+const TintUIImage = ({ tintColor, style, onLoadEnd, onError, ...rest }: UIImageProps) => {
+    const theme = useTheme();
+    const tintColorValue: ColorValue | null = tintColor != null ? theme[tintColor] : null;
+
+    const idRef = React.useRef(nanoid());
+    const [hasError, setHasError] = React.useState<boolean>(false);
+
+    const source = rest.source as any;
+
+    const { uri } = source;
+    const { width, height } = useImageDimensions(style, source);
+
+    const [dimensions, setDimensions] = React.useState<LayoutRectangle | null>(null);
+
+    const onLayout = (event: LayoutChangeEvent) => {
+        if (rest.onLayout) {
+            rest.onLayout(event);
+        }
+        setDimensions(event.nativeEvent.layout);
+    };
+
+    React.useEffect(() => {
+        if (!tintColorValue || !width || !height || !uri) {
+            setHasError(true);
+            return;
+        }
+        var img = new Image();
+        var currentCanvas = document.getElementById(`${idRef.current}`) as any;
+        if (!currentCanvas || !currentCanvas.getContext) {
+            setHasError(true);
+            return;
+        }
+        var ctx: CanvasRenderingContext2D | undefined = currentCanvas.getContext('2d');
+        if (!ctx) {
+            setHasError(true);
+            return;
+        }
+        img.onload = () => {
+            if (!ctx || !dimensions) {
+                return;
+            }
+            // draw image
+            ctx.drawImage(img, 0, 0, dimensions.width, dimensions.height);
+            // set composite mode
+            ctx.globalCompositeOperation = 'source-in';
```

Comment:
Oh, I'm starting to recall such "blend"mode names (was useful when creating the photo editor) 😄👍